src/components/tables/RolesTable.vue

```diff
@@ -70,10 +77,10 @@ async function getRolesWithType() {
     }
 
     return role;
-  }));
+  });
 }
 
-onMounted(async () => {
-  roles.value = await getRolesWithType();
+watch(() => props.roles, () => {
+  roleList.value = getRolesWithType(props.roles);
```

Comment:
not good, have to be done in the parent component.

Reply:
done